src/store/chat/saga.ts

```diff
@@ -105,6 +104,19 @@ export function* isMemberOfActiveConversation(activeConversationId) {
   return conversationList.includes(activeConversationId);
 }
 
+// NOTE: we're waiting for the event to be fired, but if it doesn't..then
+// we should just keep showing the loading spinner?
+export function* setWhenUserJoinedRoom(conversationId: string) {
+  const { userJoined } = yield race({
```

Comment:
What happens if we've already joined the room?

Reply:
> What happens if we've already joined the room?

This function won't be called. We call it after calling the `joinRoom` (API), which is only called when user isn't in a room, or the room does not exist.

Reply:
> Also, what if it's a different room event?

not sure what you mean here? For both creation & join we get this event (since the user is invited to the room either way)

Reply:
But what if the event came in _before_ we got the result from the server?

Reply:
> > Also, what if it's a different room event?
> 
> not sure what you mean here? For both creation & join we get this event (since the user is invited to the room either way)

Sorry, I meant, what if we get a join event for a different room? Then this would still cause problems, right?

Reply:
ohh..right. we need to ensure that we get the event for "this" room, and if we get another `joinRoom` event, then we should still keep waiting. 🤔

Reply:
updated ^